server/utils/match/spaceSearch.ts

```diff
@@ -0,0 +1,116 @@
+import { Cas1SpaceSearchParameters, PlacementCriteria, PlacementRequestDetail } from '@approved-premises/api'
+import { CheckBoxItem, RadioItem } from '@approved-premises/ui'
+import { filterByType } from '../utils'
+import {
+  ApTypeCriteria,
+  apType,
+  apTypeCriteriaLabels,
+  applyApTypeToAssessApType,
+  placementCriteriaLabels,
+} from '../placementCriteriaUtils'
+import { convertKeyValuePairToRadioItems } from '../formUtils'
+
+export const spaceSearchCriteriaApLevelLabels = {
+  acceptsSexOffenders: 'Sexual offences against adults',
+  acceptsChildSexOffenders: 'Sexual offences against children',
+  acceptsNonSexualChildOffenders: 'Non sexual offences against children',
+  isSuitableForVulnerable: 'Vulnerable to exploitation (removes ESAP APs)',
+  isCatered: 'Catered',
+} as const
+
+export type SpaceSearchApCriteria = keyof typeof spaceSearchCriteriaApLevelLabels
+
+export const spaceSearchCriteriaRoomLevelLabels = {
+  isWheelchairDesignated: 'Wheelchair',
+  isStepFreeDesignated: 'Step-free',
+  hasEnSuite: 'En-suite',
+  isSingle: 'Single room',
+  isArsonSuitable: 'Arson room',
+  isSuitedForSexOffenders: 'Suitable for sexual offences',
+} as const
+
+export type SpaceSearchRoomCriteria = keyof typeof spaceSearchCriteriaRoomLevelLabels
+
+export const spaceSearchResultsCharacteristicsLabels = {
+  ...spaceSearchCriteriaApLevelLabels,
+  ...spaceSearchCriteriaRoomLevelLabels,
+  isSuitableForVulnerable: placementCriteriaLabels.isSuitableForVulnerable,
+} as const
+
+export type SpaceSearchState = {
+  applicationId: string
```

Comment:
Why do we need the application ID in here? The search state is always keyed on the placementRequest Id, so I would have thought the application ID would be available through that.

Reply:
Not strictly required, but it allows us to simply pass the search state to produce the parameters needed when making the actual search call -- so we can either have the `applicationId` as part of the state, or extend the signature for performing the search for both the client and service to also include an `applicationId` parameter.